pkg/agent/endpoints/sdsv3/handler.go

```diff
@@ -142,6 +143,10 @@ func (h *Handler) StreamSecrets(stream secret_v3.SecretDiscoveryService_StreamSe
 						telemetry.Expect:      versionInfo,
 					}).Error("Client rejected expected version and rolled back")
 				}
+				// If Envoy doesn't return Node, use the Node from a previous stream
+				if newReq.Node == nil && lastNode != nil {
+					newReq.Node = lastNode
+				}
```

Comment:
Small suggestion to the comment and simplification of the conditional.
```suggestion
				// If the current request does not contain node information, use the information from a previous request (if any)
				if newReq.Node == nil {
					newReq.Node = lastNode
				}
```